tianshou/data/collector.py

```diff
@@ -282,7 +282,11 @@ def collect(self,
             for j, i in enumerate(self._ready_env_ids):
                 # j is the index in current ready_env_ids
                 # i is the index in all environments
-                self._cached_buf[i].add(**self.data[j])
+                if self.buffer is not None:
+                    self._cached_buf[i].add(**self.data[j])
+                else:
+                    self._cached_buf[i].add(
+                        obs=0, act=0, rew=self.data.rew[j], done=0)  # fakedata
```

Comment:
Is it better to change the initial behavior of `self._cache_buf`?
```python
self._cache_buf = [] if buffer is None else [ListReplayBuffer() for i in range(self.env_num)]
```

Reply:
I think storing fake data is ok and memory efficient. Making no ``_cache_buf `` will require more if-else, which I personally dislike. The ``collect`` already has many if-else (async/sync, n_step/total_n_episode/exact_n_episodes).